### DISCUSSION TITLE
Training synthetic data for real life objects

### DISCUSSION THREAD

**Discussion by kvasig** [ultralytics/yolov5#7238] [Q&A]
Hi! Currently trying yolov5 with custom synthetic data. The dataset we've created consists of 8 different objects. Each object has a minimum of 1500 pictures/labels, where the pictures are split 500/500/500 of normal/fog/distractors around object. Sample images from the dataset is in the first imgur link. The model is not trained from scratch, but from yolov5 standard .pt. 

So far we've tried:
Adding more data (from 300 images per object, to 4500)
Creating more complex data (distractors on/around objects)
Running multiple runs of training
Trained with model size small, medium, large, xlarge
Different batch size between 4-32 (depending on model size)

Everything so far has resulted in good/great detection on synthetic data, but completely off when used on real-life data. 
Examples: Thinks that the whole pictures of unrelated objects is a paperbox, walls are pallets, etc. 

Anyone got clues for how to improve the training or data to be better suited for real life detection? Or how to better interpret the results?

Results uploaded to imgur: 
https://imgur.com/a/P0TQeBl

Example on real life data:
https://imgur.com/a/SGY7w8w

**Comment by glenn-jocher** [comment]
@kvasig synthetic data is not recommended. For best results training space and deployment space must both be subsets of a larger common image space. If they diverge then your model will fail to generalize, which is the (expected) behavior you are seeing.

![Screenshot 2021-08-04 at 13 15 05](https://user-images.githubusercontent.com/26833433/161331029-ede00027-ff24-47ae-8ee4-fcb66747295b.jpg)
